### PR TITLE
Stage3

### PR DESCRIPTION
**Pull request by AZhurav1ev**

### CODE REVIEW

File: sources/views/contacts.js [hunk 1]
```diff
@@ -62,4 +72,10 @@ export default class Contacts extends JetView {
 		this.list.unselectAll();
 		this.show("/top/contacts/contactsForm");
 	}
+
+	contactsFilter() {
+		const value = this.$$("contactsFilterInput").getValue().toLowerCase();
+		this.list.filter(contact => contact.value.toLowerCase().indexOf(value) !== -1 ||
+			contact.Job.toLowerCase().indexOf(value) !== -1);
```
Comment (roman-zaitsev) [inline sources/views/contacts.js]: Required features:
filtering contacts by all data attributes from the one input

I understood that contacts should be filtered not only by these two attributes but by all of them.

File: sources/views/settingsForm.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			if (itemData.type === "Status" && itemData.id) {
+				statuses.updateItem(itemData.id, itemData);
+			}
+			if (itemData.type === "Activity" && !itemData.id) {
+				activitiesTypes.add(itemData, 0);
+			}
+			if (itemData.type === "Status" && !itemData.id) {
+				statuses.add(itemData, 0);
+			}
+		}
```
Comment (roman-zaitsev) [inline sources/views/settingsForm.js]: Now if form not validated it closed in any case, so nothing happens. It should show validate message and not close the window.

File: sources/views/contacts.js (hunk 1 above)
Comment (AZhurav1ev) [reply sources/views/contacts.js]: Added all fields from contacts details to filter function.

File: sources/views/settingsForm.js (hunk 2 above)
Comment (AZhurav1ev) [reply sources/views/settingsForm.js]: Changed closing form function position.

File: sources/myapp.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default class MyApp extends JetApp {
 			version: VERSION,
 			router: BUILD_AS_MODULE ? EmptyRouter : HashRouter,
 			debug: !PRODUCTION,
-			start: "/top/contacts"
+			start: "/top/settings"
```
Comment (HelgaListopad) [inline sources/myapp.js]: Suitable to demonstrate the stage3, but not for the app in general. Leave Contacts as the start page.

File: sources/views/contacts.js [hunk 4]
```diff
@@ -62,4 +72,21 @@ export default class Contacts extends JetView {
...
+		this.list.filter((contact) => {
+			if (
+				contact.value.toLowerCase().indexOf(value) !== -1 ||
+				contact.Email.toLowerCase().indexOf(value) !== -1 ||
+				contact.Skype.toLowerCase().indexOf(value) !== -1 ||
+				contact.Job.toLowerCase().indexOf(value) !== -1 ||
+				contact.Company.toLowerCase().indexOf(value) !== -1 ||
+				webix.i18n.longDateFormatStr(contact.Birthday).toLowerCase().indexOf(value) !== -1 ||
+				contact.Address.toLowerCase().indexOf(value) !== -1) {
+				return contact;
```
Comment (roman-zaitsev) [inline sources/views/contacts.js]: If only these keys exists in contact you can use shorter version

return Object.keys(contact).some((key) => contact[key].indexOf(value) !== -1);

Comment (roman-zaitsev) [reply sources/views/contacts.js]: also add one if statement for birthday

Comment (AZhurav1ev) [reply sources/views/contacts.js]: Contact have more options, like Photo, Phone, StatusId etc.

Comment (roman-zaitsev) [reply sources/views/contacts.js]: return ["value", "Email", "Skype", "Job", "Company", "Address"].some((key) => contact[key].indexOf(value) !== -1) || webix.i18n.longDateFormatStr(contact.Birthday).toLowerCase().indexOf(value) !== -1;

File: sources/myapp.js (hunk 3 above)
Comment (AZhurav1ev) [reply sources/myapp.js]: Changed start page to "contacts"

**Pull request merged**